### ISSUE TITLE
has been fixed in mt6853 

### ISSUE THREAD

**Issue by CoolDUp** [chaosmaster/bypass_payloads#7]
brom dump here
[https://github.com/CoolDUp/MTK-brom-dump/blob/main/mt6853.dat](url)

**Comment by chaosmaster** [comment]
Thanks, I'll take a look.
May I ask, how you dumped the brom?

**Comment by sarunelis** [comment]
Seems command E0 (E8) is modified, after unsuccessfully data validation it clears (fills with 0x44) 0x100A00 Buffer :(

**Comment by sarunelis** [comment]
ROM:0000F21E FF F7 CB FB                 BL              comdl_get_data_sendcks_sub_E9B8
ROM:0000F222 49 48                       LDR             R0, =EXPL_unk_100A00
ROM:0000F224 01 21                       MOVS            R1, #1
ROM:0000F226 F7 F7 55 FB                 BL              Validate_E0_sub_68D4
ROM:0000F22A 04 46                       MOV             R4, R0
ROM:0000F22C FF 2C                       CMP             R4, #0xFF
ROM:0000F22E 04 D3                       BCC             loc_F23A
ROM:0000F230 45 48                       LDR             R0, =EXPL_unk_100A00
ROM:0000F232 44 21                       MOVS            R1, #0x44 ; 'D'
ROM:0000F234 32 46                       MOV             R2, R6
ROM:0000F236 02 F0 1D FA                 BL              MEM_FILL_sub_11674    <<<<<<<<<<<<<<<
ROM:0000F23A
ROM:0000F23A             loc_F23A                                ; CODE XREF: DLCMD_E0_E8_sub_F1D0+28j
ROM:0000F23A                                                     ; DLCMD_E0_E8_sub_F1D0+2Ej
ROM:0000F23A                                                     ; DLCMD_E0_E8_sub_F1D0+3Ej
ROM:0000F23A                                                     ; DLCMD_E0_E8_sub_F1D0+5Ej
ROM:0000F23A A0 B2                       UXTH            R0, R4
ROM:0000F23C 01 21                       MOVS            R1, #1
ROM:0000F23E FF F7 5E FB                 BL              comdl_put_word_sub_sub_E8FE

**Comment by chaosmaster** [comment]
It was also fixed in the usb handler itself.
Would still be interested in how brom was dumped @CoolDUp 
Was it an insecure device? If so, which one?

**Comment by awsaxf** [comment]
Dimensity 720 and Dimensity 800U are also MT6853, but I think they should be different!Dimensity 800U, testing var_1 to 0x2c8 still can not dump brom! Dimensity 1200 (MT6893) test to 0x10b can not dump brom!

**Comment by chaosmaster** [comment]
_var_1_ for MT6853 would've been _0xA_ if it wasn't fixed.
If I had to guess, MT6893 will probably also be fixed.

**Comment by awsaxf** [comment]
Yes, but I still want to try again, I bought a few machines to test!

**Comment by chaosmaster** [comment]
> Yes, but I still want to try again, I bought a few machines to test!

Good luck!
Hopefully one of your devices has security disabled and allows dumping the bootrom.

**Comment by k25c2yf** [comment]
> > Yes, but I still want to try again, I bought a few machines to test!
> 
> Good luck!
> Hopefully one of your devices has security disabled and allows dumping the bootrom.

![QQ图片20210701021739](https://user-images.githubusercontent.com/22695577/124029298-9434cc80-da27-11eb-934d-6a343cef1328.png)

I log in to the original download software and can use realme Q2 (MT6583),anything I can do to help?

download:
https://drive.google.com/file/d/1ksZBNZJVJDUOAsSv-aafeCbnAzVaXPVn/view?usp=sharing

Note: The account may expire or be cancelled soon, please try to detect usb data/dump the certificate as soon as possible.


This is a Wireshark usb flashing dump and updataing log file.
https://drive.google.com/file/d/1tJyypd0L6yGFTj1JCGPR6RKu6oe8KgsF/view?usp=sharing

log
https://drive.google.com/file/d/1mgHCT5cgxpo-7qPtX3UAfa-XjDLs2jU7/view?usp=sharing

**Comment by sarunelis** [comment]
Its Ok and have nothing todo with BROM exploit. Maybe you not understoond full picture of BBK flashing process:
After DA is loaded, DA need extra authenticaction with BBK server to continue flashing process (Its like VIP on SnapDragon)

**Comment by awsaxf** [comment]
I found that Chimera tools are already supported, how did they do it?[https://chimeratool.com/zh/download](url)

**Comment by awsaxf** [comment]
[mt6853_payload.zip](https://github.com/chaosmaster/bypass_payloads/files/6801699/mt6853_payload.zip)
Hello~This is the payload file grabbed from bypass. I don’t know the address, and it prompts "Payload did not reply"

**Comment by victoreduardob2k** [comment]
Someone already have a bypass for mt6853 ? I need that :(

**Comment by victory789** [comment]
> Someone already have a bypass for mt6853 ? I need that :(

I also found that some tools already support all Dimensity SOC

**Comment by victoreduardob2k** [comment]
@victory789  it's a free tools ? 
If it works , please give link 
Thanks

**Comment by victory789** [comment]
> @victory789 it's a free tools ?
> If it works , please give link
> Thanks

https://chimeratool.com/    
This is the link I found in Google. I learned about it. It should be an annual fee.

**Comment by danis1233** [comment]
![20210812_002105](https://user-images.githubusercontent.com/86980997/129105219-84adbe4e-cbf5-43dd-930b-65263585d035.jpg)
what is wrong?  dimensity 800u, mt6853v/t, realme 7 5g

**Comment by chaosmaster** [comment]
Is UsbDK installed?

**Comment by danis1233** [comment]
UsbDK instaled, but:
![IMG_20210812_094336](https://user-images.githubusercontent.com/86980997/129150273-0bc86bf3-0dbb-4550-9fc5-961f8f3e305f.jpg)
UsbDK driver instalation failed, its ok?

**Comment by chaosmaster** [comment]
UsbDk is required for the new SOCs like 6853.
Alternatively you can use Linux which should be much more reliable.

**Comment by danis1233** [comment]
how to check whether the UsbDK is installed correctly?

**Comment by sarunelis** [comment]
https://github.com/daynix/UsbDk

**Comment by MofadalAltyeb** [comment]
for @chaosmaster libusb-0 can be used as well for MT6853,MT6833,MT6891Z,MT6893,
 i used in custom (cpp) software, its based on your exploit,
 works perfect also under win 11, 
old libusb is simplified, less issues on different windows version,
the thing is you need to reset_ep/release_ifc and close the device after every control transfer operation,
 then repeat this open/reset/close,
also need to get line coding data just right before update/register (not sure maybe enough to get at the beginning).
here exmaple code i used for libusb0 for new soc's:
bool BRomUpdateLC(uint32_t addr)
{
    qbyte get_linecode = {0x7, Qt::Uninitialized};
    if(!UsbSendDevCtrl(0xa1, 0x21, 0x00, 0x00, get_linecode))
        return 0;

    qbyte data= {};
    data.append(get_linecode);//7 b
    data.append(mCallback::GetNum(0x00));//00
    data.append(mCallback::GetLe32(addr));//addr_le
    data.append(mCallback::GetNum(0x00)); //00

//data = (line_coding + addr_le) will not work
    if(!UsbSendDevCtrl(0x21, 0x20, 0x00, 0x00, data))
        return 0;

    qbyte reset = {0x7, Qt::Uninitialized};
    if(!UsbSendDevCtrl(0x80, 0x06, 0x0200, 0x00, reset))
        return 0;

    return 1;
}
UsbSendDevCtrl =
usb_open
usb_control_msg(wLength and data filed's very sensitive for (return data case))
usb_resetep or usb_release_interface (usb_resetep for win -10)
usb_close

i have tried everything fails but this is the only method which works fine for me.
